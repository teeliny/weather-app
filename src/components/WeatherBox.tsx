import React from 'react';
import Box from '@mui/material/Box';
import styled from '@emotion/styled';
import { convertTemp } from '../utils/basicFormatter';

const InnerDiv = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

interface IWeatherBox {
  temp: number;
  humidity: number;
  current_date: string;
  tempUnit: string;
  id: string;
  handleSelectDay: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
}

function WeatherBox({temp, humidity, current_date, tempUnit, id, handleSelectDay}: IWeatherBox) {
  return (
    <Box
      id={id}
      sx={{
        width: 120,
        height: 120,
        margin: '0 auto',
        padding: '1rem 2rem',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between',
        alignItems: 'center',
        border: '1px solid #000000',
      }}
      onClick={handleSelectDay}
    >
      <p>Temperature</p>
      <InnerDiv>
        <p>{(convertTemp(tempUnit, temp)).toFixed(2)} {tempUnit === '0' ? 'C' : 'F'}</p>
        <p>{humidity.toFixed(0)}</p>
      </InnerDiv>
      <p>{current_date}</p>
    </Box>
  )
}

export default WeatherBox;
