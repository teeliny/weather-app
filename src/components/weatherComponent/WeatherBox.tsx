import React from 'react';
import Box from '@mui/material/Box';
import { convertTemp } from '../../utils/formatter/basicFormatter';
import { CardWrapper } from './weather.style';
import { IWeatherBox } from './weather.typing';

function WeatherBox({temp, humidity, current_date, tempUnit, id, handleSelectDay}: IWeatherBox) {
  return (
    <Box
      id={id}
      sx={{
        width: 120,
        height: 120,
        margin: '0 auto',
        padding: '1rem 2rem',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between',
        alignItems: 'center',
        border: '1px solid #000000',
      }}
      onClick={handleSelectDay}
    >
      <p>Temperature</p>
      <CardWrapper>
        <p>{(convertTemp(tempUnit, temp)).toFixed(2)} {tempUnit === '0' ? 'C' : 'F'}</p>
        <p>{humidity.toFixed(0)}</p>
      </CardWrapper>
      <p>{current_date}</p>
    </Box>
  )
}

export default WeatherBox;
