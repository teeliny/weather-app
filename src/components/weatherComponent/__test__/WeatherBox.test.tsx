import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import WeatherBox from '../WeatherBox';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <WeatherBox
      temp={30}
      humidity={60}
      current_date={''}
      tempUnit={''}
      id={''}
      handleSelectDay={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
        throw new Error('Function not implemented.');
      }}
    />,
    div,
  );
});

it('renders box correctly', () => {
  const { getByTestId } = render(
    <WeatherBox
      temp={30}
      humidity={60}
      current_date={''}
      tempUnit={''}
      id={''}
      handleSelectDay={
        (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
          throw new Error('Function not implemented.');
        }
      }
    />
  );
  expect(getByTestId('single-box')).toHaveTextContent('Temperature');
});

it('matches weather box snapshot', () => {
  const tree = renderer
    .create(
      <WeatherBox
        temp={30}
        humidity={60}
        current_date={''}
        tempUnit={'0'}
        id={''}
        handleSelectDay={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
          throw new Error('Function not implemented.');
        }}
      />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});