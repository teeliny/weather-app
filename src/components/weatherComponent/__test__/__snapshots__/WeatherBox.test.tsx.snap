// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches weather box snapshot 1`] = `
<div
  className="MuiBox-root css-l3v7cb"
  data-testid="single-box"
  id=""
  onClick={[Function]}
>
  <p>
    Temperature
  </p>
  <div
    className="css-hdtzot"
  >
    <p>
      30.00
       
      C
    </p>
    <p>
      60
    </p>
  </div>
  <p>
    
  </p>
</div>
`;
